### PR TITLE
Add formatted output generation on top of BufferWriter.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**

### CODE REVIEW

File: lib/ts/bwprint.cc [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    if (spec._radix_lead_p)
+      prefix = "0b"_sv;
+    n        = detail::bwf_to_radix<2>(i, buff, sizeof(buff), detail::LOWER_DIGITS);
+    break;
+  case 'B':
+    if (spec._radix_lead_p)
+      prefix = "0B"_sv;
+    n        = detail::bwf_to_radix<2>(i, buff, sizeof(buff), detail::UPPER_DIGITS);
+    break;
+    break;
```
Comment (PSUdaemon) [inline lib/ts/bwprint.cc]: Extra break?

File: lib/ts/bwprint.cc [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+  if (neg_p)
+    neg = "-"_sv;
+  else if (spec._sign != '-')
+    neg = string_view{&spec._sign, 1};
+  switch (spec._type) {
+  case 'x':
+    if (spec._radix_lead_p)
+      prefix = "0x"_sv;
+    n        = detail::bwf_to_radix<16>(i, buff, sizeof(buff), detail::LOWER_DIGITS);
```
Comment (PSUdaemon) [inline lib/ts/bwprint.cc]: Putting braces around the body of the `if` will fix this weird formatting and also match our style guide.

File: lib/ts/bwprint.cc (hunk 1 above)
Comment (SolidWallOfCode) [reply lib/ts/bwprint.cc]: Dang, I thought I'd fixed that.

File: lib/ts/bwprint.cc (hunk 2 above)
Comment (SolidWallOfCode) [reply lib/ts/bwprint.cc]: Haha, yeah, that's some serious ugly there, isn't it?

File: lib/ts/bwprint.cc [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+      // precision
+      if ('.' == *sz) {
+        n = tv_to_positive_decimal(++sz, &num);
+        if (num) {
+          _prec = static_cast<decltype(_prec)>(n);
+          sz.remove_prefix(num.size());
+          if (!sz)
+            return;
+        } else {
+          throw std::invalid_argument("Precision mark without precision");
```
Comment (ywkaras) [inline lib/ts/bwprint.cc]: Generic use of exceptions is questionable in a source base that has not be upgraded to exception safety.

File: lib/ts/bwprint.cc [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+      if (spec._idx < 0)
+        gf = detail::bwf_global_table_find(spec._name);
+      _items.emplace_back(spec, gf);
+    }
+  }
+}
+
+ts::BWFormat::~BWFormat()
+{
+}
```
Comment (ywkaras) [inline lib/ts/bwprint.cc]: Why?  Just, why?

Comment (SolidWallOfCode) [reply lib/ts/bwprint.cc]: Habit.

**Comment by SolidWallOfCode** [comment]
I really don't like the duplication between `bwprint` and `BWFormat::BWFormat`. I need to figure out how to unify those two things.

**Comment by SolidWallOfCode** [comment]
[approve ci]

**Pull request closed without merging**